python-spec/src/somacore/types.py

```diff
@@ -57,6 +57,16 @@ def stop(self) -> Optional[_T_co]: ...
     @property
     def step(self) -> Optional[_T_co]: ...
 
+    if sys.version_info >= (3, 12):
+
+        @property
+        def __orig_bases__(self) -> Any:
+            # Only exists in Python 3.12 and above
+            from types import get_original_bases
+
+            return get_original_bases(int)
+            # return get_original_bases(_T_co)
```

Comment:
This is a typeguard bug—it’s checking too many built-in attributes of the protocol that don’t actually matter. (See, for example, [a case where it rejects objects with a completely empty protocol](https://github.com/agronholm/typeguard/issues/479#issuecomment-2356689989) that I found.

That said, if implementing this makes typeguard stop complaining for now, I would do two things:

1. Add a comment to that effect, and that it should be removed at some point
2. Make this function instead raise `AssertionError`, since it should never be called.